### PR TITLE
Fixes #4681 and a few Windows PgSQL build problems

### LINKED ISSUE #4681: Postgres cartridge build fails under Ubuntu
**Describe the bug**

I have trouble building Postgres cartridge under Linux. All other parts of RDKit compile and pass tests (with -DRDK_BUILD_PGSQL=OFF, Windows compilation of cartridge also works)

The problem is that cartridge build fails due to invalid link command:  `/BUILD/boost/lib/libboost_iostreams-mt-d-x64.so -loptimized_static /BUILD/boost/lib/libboost_regex-mt-x64.so -ldebug_static /BUILD/boost/lib/libboost_regex-mt-d-x64.so ../../GraphMol/MolAlign/libRDKitMolAlign_static.a`

It seems that whatever code automatically adds boost libraries is somehow mangling in configuration information. I also get both debug and release versions of boost libraries in link command `../../Catalogs/libRDKitCatalogs_static.a /BUILD/boost/lib/libboost_serialization-mt-x64.so /BUILD/boost/lib/libboost_serialization-mt-d-x64.so /usr/lib/x86_64-linux-gnu/libfreetype.so` though I'm not sure if this is an issue also.

**To Reproduce**

I'm building RDKit from source using boost 1.76 built locally. Postgres 12 is installed from PostgreSQL Apt Repository. Cmake 3.21.3 is installed from KitWare apt repository.

Run cmake referencing custom boost:

cmake -DBOOST_ROOT:PATHNAME=$BOOST_BASE_PATH -DBoost_LIBRARY_DIRS=$BOOST_LIBRARYDIR -DBoost_NO_SYSTEM_PATHS=ON -DBoost_NO_BOOST_CMAKE=TRUE -DBoost_ARCHITECTURE=-x64 -DRDK_BUILD_PYTHON_WRAPPERS=ON -DPYTHON_EXECUTABLE:FILEPATH=$PYTHONBIN_DIR -DPYTHON_LIBRARY=$PYTHON_LIBRARY -DPYTHON_INCLUDE_DIR=$PYTHON_INCLUDE_DIR -DPYTHON_NUMPY_INCLUDE_PATH=$PYTHON_NUMPY_INCLUDE_PATH -DRDK_BUILD_INCHI_SUPPORT=ON -DRDK_BUILD_AVALON_SUPPORT=ON -DRDK_BUILD_FREESASA_SUPPORT=ON -DRDK_BUILD_YAEHMOP_SUPPORT=ON -DRDK_BUILD_MOLINTERCHANGE_SUPPORT=ON -DRDK_BUILD_PGSQL=ON -DPostgreSQL_CONFIG=$POSTGRES_CONFIG ..

Run make 

**Expected behavior**

Cartridge is built successfully.

**Configuration (please complete the following information):**
 - RDKit version: 2021.09.2
 - OS: Ubuntu 20.04
 - Python version : Python 3.8
 - Building rdkit from source

**Additional context**

Link.txt generated by CMake:

`/usr/bin/c++ -fPIC  -mpopcnt -Wno-deprecated -Wno-unused-function -fno-strict-aliasing -Wall -Wextra -fPIC -O3 -DNDEBUG -shared -Wl,-soname,librdkit.so -o librdkit.so CMakeFiles/rdkit.dir/adapter.cpp.o CMakeFiles/rdkit.dir/bfp_op.c.o CMakeFiles/rdkit.dir/cache.c.o CMakeFiles/rdkit.dir/guc.c.o CMakeFiles/rdkit.dir/low_gist.c.o CMakeFiles/rdkit.dir/mol_op.c.o CMakeFiles/rdkit.dir/rdkit_gist.c.o CMakeFiles/rdkit.dir/bfp_gist.c.o CMakeFiles/rdkit.dir/bfp_gin.c.o CMakeFiles/rdkit.dir/bitstring.c.o CMakeFiles/rdkit.dir/rdkit_io.c.o CMakeFiles/rdkit.dir/rxn_op.c.o CMakeFiles/rdkit.dir/sfp_op.c.o   -L/BUILD/rdkit/lib  -Wl,-rpath,/BUILD/rdkit/lib:/BUILD/boost/lib /usr/lib/x86_64-linux-gnu/libpq.so ../../../External/AvalonTools/libRDKitAvalonLib_static.a ../../../External/AvalonTools/libRDKitavalon_clib_static.a ../../../External/INCHI-API/libRDKitRDInchiLib_static.a ../../../External/INCHI-API/libRDKitInchi_static.a ../../GraphMol/MolInterchange/libRDKitMolInterchange_static.a ../../GraphMol/MolDraw2D/libRDKitMolDraw2D_static.a ../../GraphMol/MolTransforms/libRDKitMolTransforms_static.a ../../GraphMol/MolHash/libRDKitMolHash_static.a ../../GraphMol/FMCS/libRDKitFMCS_static.a ../../GraphMol/ChemReactions/libRDKitChemReactions_static.a ../../GraphMol/ChemTransforms/libRDKitChemTransforms_static.a ../../GraphMol/FileParsers/libRDKitFileParsers_static.a ../../GraphMol/SmilesParse/libRDKitSmilesParse_static.a ../../GraphMol/Fingerprints/libRDKitFingerprints_static.a ../../GraphMol/Subgraphs/libRDKitSubgraphs_static.a ../../GraphMol/Descriptors/libRDKitDescriptors_static.a ../../GraphMol/PartialCharges/libRDKitPartialCharges_static.a ../../GraphMol/Substruct/libRDKitSubstructMatch_static.a ../../GraphMol/libRDKitGraphMol_static.a ../../Numerics/EigenSolvers/libRDKitEigenSolvers_static.a ../../DataStructs/libRDKitDataStructs_static.a ../../GraphMol/Depictor/libRDKitDepictor_static.a ../../Geometry/libRDKitRDGeometryLib_static.a ../../RDGeneral/libRDKitRDGeneral_static.a ../../../External/RingFamilies/libRDKitRingDecomposerLib_static.a ../../GraphMol/FilterCatalog/libRDKitFilterCatalog_static.a ../../Catalogs/libRDKitCatalogs_static.a /BUILD/boost/lib/libboost_serialization-mt-x64.so /BUILD/boost/lib/libboost_serialization-mt-d-x64.so /usr/lib/x86_64-linux-gnu/libfreetype.so ../../GraphMol/Subgraphs/libRDKitSubgraphs_static.a ../../GraphMol/FileParsers/libRDKitFileParsers_static.a /usr/lib/x86_64-linux-gnu/libz.so ../../GraphMol/SmilesParse/libRDKitSmilesParse_static.a ../../GraphMol/Depictor/libRDKitDepictor_static.a ../../../External/CoordGen/libRDKitmaeparser_static.a /BUILD/boost/lib/libboost_system-mt-x64.so /BUILD/boost/lib/libboost_system-mt-d-x64.so /BUILD/boost/lib/libboost_iostreams-mt-x64.so /BUILD/boost/lib/libboost_iostreams-mt-d-x64.so -loptimized_static /BUILD/boost/lib/libboost_regex-mt-x64.so -ldebug_static /BUILD/boost/lib/libboost_regex-mt-d-x64.so ../../GraphMol/MolAlign/libRDKitMolAlign_static.a ../../GraphMol/MolTransforms/libRDKitMolTransforms_static.a ../../Numerics/EigenSolvers/libRDKitEigenSolvers_static.a ../../GraphMol/Substruct/libRDKitSubstructMatch_static.a ../../GraphMol/libRDKitGraphMol_static.a ../../Numerics/Alignment/libRDKitAlignment_static.a ../../Geometry/libRDKitRDGeometryLib_static.a ../../DataStructs/libRDKitDataStructs_static.a ../../RDGeneral/libRDKitRDGeneral_static.a ../../../External/CoordGen/libRDKitcoordgen_static.a -lpthread`

### PR DESCRIPTION
**Pull request by ptosco**
This PR fixes #4681. The problem is caused by the fact that the special `optimized`, `debug` (and eventually `general`) `cmake` keywords are passed to `target_link_libraries` as if they were libraries causing a linker error. The correct thing to do is skipping the keyword and makes sure to link the library that follows the keyword only if it relevant to the `CMAKE_BUILD_TYPE` that is being built.
Incidentally I also noted that if the build fails half way and then is restarted, the RDKit headers are not included when building the PostgreSQL cartridge, so I added an `include_dir` to fix the problem.
While testing the PR on Windows and attempting to link against PostgreSQL 14 I also found a couple other issues:
* on Windows PostgreSQL uses a hacky way to redefine the `stat` structure. However, the hack relies on `sys/stat.h` being included by `win32_port.h`. As in `adapter.c` `boost` includes that header earlier, when `win32_port.h` attempts to include it the hack fails. The fix is including it earlier using the same hack used by PostgreSQL. The hack is harmless as `adapter.c` does not use the `stat` structure
* The Windows install script had a typo due to the quote being in the wrong place and the separator being a forward slash rather than a backslash

### CODE REVIEW

File: Code/PgSQL/rdkit/adapter.cpp [hunk 1]
```diff
@@ -30,6 +30,17 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
+
+// PostgreSQL 14 on Windows uses a hack to redefine the stat struct
+// The hack assumes that sys/stat.h will be imported for the first
+// time by win32_port.h, which is not necessarily the case
+// So we need to set the stage for the hack or it will fail
+#ifdef _WIN32
```
Comment (greglandrum) [inline Code/PgSQL/rdkit/adapter.cpp:38]: is this always true for WIN32 or only when you are using MSVC, in which case it probably should be:
```suggestion
#ifdef _MSC_VER
```

File: Code/PgSQL/rdkit/adapter.cpp [hunk 2]
```diff
@@ -76,6 +87,12 @@
 #include <GraphMol/MolInterchange/MolInterchange.h>
 #endif
 
+// see above comment on the PostgreSQL hack
+#ifdef _WIN32
```
Comment (greglandrum) [inline Code/PgSQL/rdkit/adapter.cpp:91]: same question about the `#ifdef` here

**Review by greglandrum — COMMENTED** [review]
I think the `#ifdef` being used probably needs to be changed, but otherwise LGTM

**Comment by ptosco** [comment]
@greglandrum I have put in `_WIN32` intentionally, because the `win32_port.h` header is included based on these pre-processor `#if`:
```
#if defined(WIN32) && !defined(__CYGWIN__)
```
which will also apply to non-MSVC Windows builds.
I admit that I have not tried to run a MinGW PostgreSQL build, but I can definitely try for peace of mind.
However, this will most likely have to wait until the weekend.

**Comment by greglandrum** [comment]
@ptosco : That makes sense. 
I don't think doing the mingw build is necessary (but I'll hold off on merging if you want to do that first).

**Comment by ptosco** [comment]
@greglandrum Yes, please wait a couple more days as I'd like to test that first. Thanks!

**Comment by ptosco** [comment]
@greglandrum Checked that I get a functional build on both MSVC and MinGW. I think I had never done a PgSQL cartridge MinGW build before - `CMakeLists.txt` didn’t really support it. This is good to go now.

**Review by greglandrum — APPROVED** [review]

**Pull request merged**